### ISSUE TITLE
Scroll position during prepend edge case (0.3.6)

### ISSUE THREAD

**Issue by aquamme** [runspired/smoke-and-mirrors#50]
Scroll position isn't maintained during a prepend if the height of the new page is much larger than that of the current page.

Suppose that there is 1000px of scrollable area. If there is 300px of content on a page when a prepend is triggered and the height of this new content is 2000px, then the required scroll position to maintain the top visible element in the original place is greater than what the viewport can fit.

In short, this case arises when transitioning from contentHeight > viewportHeight to contentHeight < viewportHeight.

**Comment by runspired** [comment]
To clarify, is this during a prepend, or when restoring scroll position during a new visit to a route?  If just prepend, this should be addressed already in `0.3.6` (observable in the `infinite-scroll` demo).

For restoring, `idForFirstItem` will generally work better than `scrollPosition` if an identical content array isn't going to be loaded.  This is an edge case that I'm going to account for in an upgrade to the `position-tracker`.  You'll be able to specify a "starting height" for the scrollable area, which will function as a min-height and start growing once enough content has been loaded to exceed it.

**Comment by runspired** [comment]
Closing in favor of #69
